File: src/main/java/com/newco/mvc/gui/MainFrame.java
package com.newco.mvc.gui;

import javax.swing.*;

public class MainFrame extends JFrame {
    public MainFrame() {
        super("Swing Demo");

        setJMenuBar(createMenu());
        setSize(1000, 800);
        setDefaultCloseOperation(DISPOSE_ON_CLOSE);
        setVisible(true);
    }

    private JMenuBar createMenu() {
        var menuBar = new JMenuBar();
        var fileMenu = new JMenu("File");
        var exitItem = new JMenuItem("Exit");

        exitItem.addActionListener(e -> System.exit(0));

        fileMenu.add(exitItem);
        menuBar.add(fileMenu);

        return menuBar;
    }
}

File: src/main/java/com/newco/mvc/gui/MainFrame.kt
package com.newco.mvc.gui

import java.awt.event.ActionEvent
import javax.swing.JFrame
import javax.swing.JMenu
import javax.swing.JMenuBar
import javax.swing.JMenuItem

class MainFrame : JFrame("Swing Demo") {
    init {
        jMenuBar = createMenu()
        setSize(1000, 800)
        defaultCloseOperation = DISPOSE_ON_CLOSE
        isVisible = true
    }

    private fun createMenu(): JMenuBar {
        val menuBar = JMenuBar()
        val fileMenu = JMenu("File")
        val exitItem = JMenuItem("Exit")
        exitItem.addActionListener { e: ActionEvent? -> System.exit(0) }
        fileMenu.add(exitItem)
        menuBar.add(fileMenu)
        return menuBar
    }
}